dev/conductor/core/lib/src/packages_autoroller.dart

```diff
@@ -126,25 +126,44 @@ This PR was generated by the automated
     if (exitCode != 0) {
       throw ConductorException('Failed to update packages with exit code $exitCode');
     }
-    // If the git checkout is clean, then pub packages are already at latest that cleanly resolve.
+    // If the git checkout is clean, then pub packages are already at latest
+    // that cleanly resolve.
     if (await framework.gitCheckoutClean()) {
       return false;
     }
     await framework.commit(
       'roll packages',
       addFirst: true,
-      author: author,
+      author: gitAuthor,
     );
     return true;
   }
 
-  @visibleForTesting
-  Future<void> generateGradleLockfiles(Directory repoRoot) async {
+  Future<void> _regenerateGradleLockfiles(Directory repoRoot) async {
     await framework.runDart(<String>[
       '${repoRoot.path}/dev/tools/bin/generate_gradle_lockfiles.dart',
       '--no-gradle-generation',
       '--no-exclusion',
     ]);
+    switch (CheckoutStatePostGradleRegeneration(await framework.gitStatus(), framework.fileSystem.path)) {
+      // If the git checkout is clean, we did not update any lockfiles and we do
+      // not need an additional commit.
+      case NoDiff():
+        return;
+      case OnlyLockfileChanges():
+        await framework.commit(
+          'Re-generate Gradle lockfiles',
+          addFirst: true,
+          author: gitAuthor,
+        );
+      case NonLockfileChanges(changes: final List<String> changes):
+        throw StateError(
+          'Expected all diffs after re-generating gradle lockfiles to end in '
```

Comment:
For my own understanding: If the roll fails, where are these failures reported and is that monitored by anyone?

Reply:
It will block the tree :) :fire: :fire_engine:

Reply:
our friend runs on post-submit for flutter/flutter: https://ci.chromium.org/ui/p/flutter/builders/luci.flutter.prod/Linux%20packages_autoroller

Reply:
Maybe an ideal future setup would be:

1. run this on a CRON job
2. failures do not block any tree, but files an issue with label `team-infra` instead

It would still have to be in prod pool though, in order to access the GitHub secrets (it can't be marked bringup).